### PR TITLE
feat: Added exception handling to SDLFacade

### PR DESCRIPTION
**Pull request by Mfrielink**

### CODE REVIEW

File: src/engine/SDLFacade.cpp [hunk 1]
```diff
@@ -183,9 +188,9 @@ namespace Engine {
 
     void SDLFacade::_add_image_in_map(string path) {
         SDL_Surface* image = IMG_Load(this->_get_absolute_path(path).c_str());
-        if (image == NULL) {    // TODO: exception
-            cout << "FAILED TO FIND THE IMAGE" << endl;
-            cout << path << endl;
+        if (image == NULL) {
+            cout << "IMG_Load: " << IMG_GetError() << endl;
+            throw Exceptions::InvalidImageException();
```
Comment (jorgdebont) [inline src/engine/SDLFacade.cpp]: exception isnt caught

File: src/engine/SDLFacade.cpp [hunk 2]
```diff
@@ -207,7 +212,10 @@ namespace Engine {
             this->_add_image_in_map(path);
         }
         int w, h;
-        SDL_QueryTexture(this->_images[path], NULL, NULL, &w, &h); // get width en height from texture
+
+        if(SDL_QueryTexture(this->_images[path], NULL, NULL, &w, &h) == 0){ // get width en height from texture
+            cout << "Draw_image: "  << SDL_GetError() << endl;
```
Comment (jorgdebont) [inline src/engine/SDLFacade.cpp]: W and H are not filled when this fails

File: src/engine/SDLFacade.cpp [hunk 3]
```diff
@@ -571,8 +579,8 @@ namespace Engine {
     {
         this->stop_music();
         this->_music = Mix_LoadMUS(this->_get_absolute_path(path).c_str());
-        if(this->_music == NULL) { //TODO exception{
-            return;
+        if(this->_music == NULL) {
```
Comment (jorgdebont) [inline src/engine/SDLFacade.cpp]: exception is never caught

File: src/engine/SDLFacade.cpp [hunk 4]
```diff
@@ -588,8 +596,8 @@ namespace Engine {
             return;
         }
         Mix_Chunk* sound_effect = Mix_LoadWAV(this->_get_absolute_path(path).c_str());
-        if (sound_effect == NULL) { //TODO exception{
-            return;
+        if (sound_effect == NULL) {
+            throw Exceptions::InvalidWavFileException();
```
Comment (jorgdebont) [inline src/engine/SDLFacade.cpp]: exception is not caught

File: src/engine/SDLFacade.cpp [hunk 5]
```diff
@@ -602,7 +610,7 @@ namespace Engine {
     void SDLFacade::play_sound_effect(const string name)
     {
         if (this->_sound_effects.find(name) == this->_sound_effects.end()) { //if sound_effect is not loaded
-            return; //TODO exception{
+            throw Exceptions::SoundeffectNotLoadedException();
```
Comment (jorgdebont) [inline src/engine/SDLFacade.cpp]: exception is not caught

File: src/engine/SDLFacade.cpp [hunk 6]
```diff
@@ -622,9 +630,10 @@ namespace Engine {
         SDL_Surface* image = IMG_Load(this->_get_absolute_path(path).c_str());
         int size = 0;
 
-        if (image == NULL) { //TODO: exception
+        if (image == NULL) {
             cout << "FAILED TO FIND THE IMAGE" << endl;
             cout << path.c_str() << endl;
+            throw Exceptions::InvalidImageException();
```
Comment (jorgdebont) [inline src/engine/SDLFacade.cpp]: exception is not caught

File: src/engine/SDLFacade.cpp (hunk 1 above)
Comment (Mfrielink) [inline src/engine/SDLFacade.cpp]: done

File: src/engine/SDLFacade.cpp (hunk 2 above)
Comment (Mfrielink) [inline src/engine/SDLFacade.cpp]: done

File: src/engine/SDLFacade.cpp (hunk 3 above)
Comment (Mfrielink) [inline src/engine/SDLFacade.cpp]: done

File: src/engine/SDLFacade.cpp (hunk 4 above)
Comment (Mfrielink) [inline src/engine/SDLFacade.cpp]: done

File: src/engine/SDLFacade.cpp (hunk 5 above)
Comment (Mfrielink) [inline src/engine/SDLFacade.cpp]: done

File: src/engine/SDLFacade.cpp (hunk 6 above)
Comment (Mfrielink) [inline src/engine/SDLFacade.cpp]: done

**Pull request closed without merging**